File: Programmers/Python/92334.py
def solution(id_list, report, k):
    declaration = {id: set() for id in id_list}
    for re in report :
        declared, reported = map(str, re.split())
        declaration[declared].add(reported)
    counts = {id: 0 for id in id_list}
    for de in declaration:
        for p in list(declaration[de]) : counts[p]+=1
    stoppedUser = set()
    [stoppedUser.add(p) for p in counts if counts[p] >= k]
    return [len(declaration[id] & stoppedUser) for id in id_list]

File: Programmers/Swift/92334.swift
import Foundation

func solution(_ id_list:[String], _ report:[String], _ k:Int) -> [Int] {
    var idCounts: [String: Int] = [:]
    for re in Set(report) {
        let p = re.split(separator: " ").map{String($0)}[1]
        idCounts.updateValue((idCounts[p] ?? 0)+1, forKey: p)
    }
    var mainCounts: [String: Int] = [:]
    for re in Set(report) {
        let p = re.split(separator: " ").map{String($0)}
        if idCounts[p[1]]! >= k {
            mainCounts.updateValue((mainCounts[p[0]] ?? 0)+1, forKey: p[0])
        }
    }
    return id_list.map { mainCounts[$0] ?? 0 }
}